src/server/frontend_xwayland/xwayland_surface.cpp

```diff
@@ -643,6 +640,28 @@ void mf::XWaylandSurface::move_resize(uint32_t detail)
     }
 }
 
+void mf::XWaylandSurface::WindowState::apply_change(
+    std::shared_ptr<XCBConnection> const& connection,
+    NetWmStateAction action,
+    xcb_atom_t net_wm_state)
+{
+    bool nil{false}, *prop_ptr = &nil;
+
+    if (net_wm_state == connection->net_wm_state_hidden)
+        prop_ptr = &minimized;
+    else if (net_wm_state == connection->net_wm_state_maximized_horz) // assume vert is also set
+        prop_ptr = &maximized;
+    else if (net_wm_state == connection->net_wm_state_fullscreen)
+        prop_ptr = &fullscreen;
+
```

Comment:
Should we be logging anything when we've asked to apply a change to a state we don't handle?

Reply:
Pre-existing.